pkg/operator/aws_platform_service_location/controller.go

```diff
@@ -38,12 +49,94 @@ func NewController(operatorClient operatorv1helpers.OperatorClient, infraClient
 }
 
 func (c AWSPlatformServiceLocationController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	currentInfra, err := c.infraClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	obj, err := c.infraLister.Get("cluster")
 	if errors.IsNotFound(err) {
-		// log not found errors, but don't degrade?
-		klog.Warningf("%w", err)
+		syncCtx.Recorder().Warningf("AWSPlatformServiceLocationController", "Required infrastructures.%s/cluster not found", configv1.GroupName)
 		return nil
 	}
-	klog.Info(spew.Sprintf("current state: %s", currentInfra.Status))
+	if err != nil {
+		return err
+	}
+
+	currentInfra := obj.DeepCopy()
+	var platformName configv1.PlatformType
+	if pstatus := currentInfra.Status.PlatformStatus; pstatus != nil {
```

Comment:
Seems like the `Status.PlatformStatus` and `Status.Platform` should always be the same.

Reply:
these should be same, but aren't currently. Migrations should make them same. and then we can drop the handling.